sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ConnectedApService.java

```diff
@@ -0,0 +1,68 @@
+// SPDX-FileCopyrightText: 2023 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.scl;
+
+import org.lfenergy.compas.scl2007b4.model.*;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.lfenergy.compas.sct.commons.util.SclConstructorHelper.newAddress;
+
+public class ConnectedApService {
+
+    public Optional<TConnectedAP> findConnectedAp(TCommunication tCommunication, String iedName, String apName) {
+        if (!tCommunication.isSetSubNetwork()) {
+            return Optional.empty();
+        }
+        return tCommunication.getSubNetwork().stream()
+                .filter(TSubNetwork::isSetConnectedAP)
+                .flatMap(tSubNetwork -> tSubNetwork.getConnectedAP().stream())
+                .filter(tConnectedAP -> iedName.equals(tConnectedAP.getIedName()) && apName.equals(tConnectedAP.getApName()))
+                .findFirst();
+    }
+
+    /**
+     * Create A GSE Section or update an existing GSE Section (the network configuration of a GSEControl block).
+     */
+    public void updateGseOrCreateIfNotExists(TConnectedAP tConnectedAP, String ldInst, String cbName, List<TP> listOfP, TDurationInMilliSec minTime, TDurationInMilliSec maxTime) {
```

Comment:
Such operation can be done by Editor suffixed service ?

Reply:
I was hesitating about that. I will move the method in the Editor service.
But I think I can extract the first statement in a findGse method and leave findGse in ConnectedApService.

Reply:
`updateGseOrCreateIfNotExists` and `updateSmvOrCreateIfNotExists`  those methods modifies SCL element under ConnectedAP section. It can take place in ConnectedApEditorService and not ControlBlockEditorService.

Reply:
I see, but `findConnectedAp`, `findGse`, and `findSmv` are also used only by `ControlBlockEditorService`, actually are public.

Reply:
I suggest using ControlBlock-related functions like `findGse` and `findSmv` as private in `ControlBlockEditorService` and remove `ConnectedApService`.
You can also rename `ControlBlockService` to `ControlService` as it handles [Control](https://com-pas.github.io/compas-sct/javadoc/org/lfenergy/compas/scl2007b4/model/TControl.html ) objects, which is something different to [ControlBlock](
https://com-pas.github.io/compas-sct/javadoc/org/lfenergy/compas/scl2007b4/model/TControlBlock.html).

Reply:
- OK to move `findGse` and `findSmv` to `ControlBlockEditorService`. But should I also move `findConnectedAp` ? Otherwise I need to keep the `ConnectedApService`.
- OK to rename `ControlBlockService` to `ControlService`.

Reply:
`findConnectedAp` is only needed to keep error message, we can add the message when dealing with  `findGse` or `findSmv`. What do you think  ?

Reply:
By removing `ConnectedApService` the `ControlBlockEditorService` will depend only on `ControlService` which make it more stable service as it doesn't depend on many other components.

Reply:
OK